src/runtime/program/vote/execute.zig

```diff
@@ -138,10 +174,287 @@ fn intializeAccount(
         clock,
     );
     defer vote_state.deinit();
-    try vote_account.serializeIntoAccountData(vote_state);
+    try vote_account.serializeIntoAccountData(VoteStateVersions{ .current = vote_state });
+}
+
+/// Agave https://github.com/anza-xyz/agave/blob/0603d1cbc3ac6737df8c9e587c1b7a5c870e90f4/programs/vote/src/vote_processor.rs#L77-L79
+fn executeAuthorize(
+    allocator: std.mem.Allocator,
+    ic: *InstructionContext,
+    vote_account: *BorrowedAccount,
+    pubkey: Pubkey,
+    vote_authorize: VoteAuthorize,
+) InstructionError!void {
+    const clock = try ic.getSysvarWithAccountCheck(
+        Clock,
+        @intFromEnum(vote_instruction.Authorize.AccountIndex.clock_sysvar),
+    );
+
+    try authorize(
+        allocator,
+        ic,
+        vote_account,
+        pubkey,
+        vote_authorize,
+        clock,
+        null,
+    );
+}
+
+/// Agave https://github.com/anza-xyz/agave/blob/0603d1cbc3ac6737df8c9e587c1b7a5c870e90f4/programs/vote/src/vote_state/mod.rs#L678
+///
+/// Authorize the given pubkey to withdraw or sign votes. This may be called multiple times,
+/// but will implicitly withdraw authorization from the previously authorized
+/// key.
+fn authorize(
+    allocator: std.mem.Allocator,
+    ic: *InstructionContext,
+    vote_account: *BorrowedAccount,
+    authorized: Pubkey,
+    vote_authorize: VoteAuthorize,
+    clock: Clock,
+    signers: ?std.AutoHashMap(Pubkey, void),
+) InstructionError!void {
+    const versioned_state = try vote_account.deserializeFromAccountData(
+        allocator,
+        VoteStateVersions,
+    );
+    var vote_state = versioned_state.convertToCurrent(allocator) catch {
+        // TODO okay to convert out of memory to InvalidAccountData?
+        return InstructionError.InvalidAccountData;
+    };
```

Comment:
i think if we OOM in the program exec we should panic - and we shouldnt do any conversions in the state methods - its equivalent to what would happen in rust

Reply:
[5b20988](https://github.com/Syndica/sig/pull/576/commits/5b20988e3e7b71b6b3267bf464732fdfcf9c203b)